poc_iot_injector/src/server.rs

```diff
@@ -111,20 +119,19 @@ async fn submit_txns(
 
     store
         .source_unordered(LOADER_WORKERS, stream::iter(file_list).map(Ok).boxed())
-        .for_each_concurrent(STORE_WORKERS, |msg| {
+        .try_for_each_concurrent(STORE_WORKERS, |msg| {
             let mut shared_txn_service = txn_service.clone();
             let shared_key = keypair.clone();
             async move {
-                match msg {
-                    Ok(m) => {
-                        handle_txn_submission(m, shared_key, &mut shared_txn_service, before_ts)
-                            .await
-                    }
-                    Err(e) => tracing::error!("unable to process msg due to: {:?}", e),
+                if let Ok(txn) =
+                    handle_txn_submission(msg, shared_key, &mut shared_txn_service, before_ts).await
+                {
+                    file_sink_write!("signed_poc_receipt_txn", receipt_sender, txn).await?;
                 }
```

Comment:
```suggestion
                handle_txn_submission(msg, shared_key, &mut shared_txn_service, before_ts).await?
                file_sink_write!("signed_poc_receipt_txn", receipt_sender, txn).await?
```
am i misunderstanding the behavior of the `?` in error handling or would the above give you the same conditional execution of the file_sink write b/c a failed txn submission would bail out to the caller first?

Reply:
It should, but for some reason doing so makes the error code angry. Not quite sure how to fix tbh.

Reply:
yeah, fair point; it's weird that it's trying to cast the error in `handle_txn_submission` _to_ the `file_store::Error` instead of the other way around; i suspect it's an oddity of the `file_sink_write!` being a macro and co-opting the way the errors typically propogate?

Reply:
maybe something to look into as a separate issue